packages/@okta/vuepress-site/guides/sign-into-mobile-app/check-for-tokens/iOS/checkfortoken.md

```diff
@@ -1 +1,52 @@
-check for token on startup code example
\ No newline at end of file
+After the `OktaOidcStateManager` instance is created, make sure that the user is signed in:
+
+```swift
+if let accessToken = authStateManager?.accessToken {
+    // use accessToken
+}
+```
+
+Please note that computed property `accessToken` validates token's expiration date. The `accessToken` propery returns `nil` if token has been expired:
+
+```swift
+@objc open var accessToken: String? {
+    // Return the known accessToken if it hasn't expired
+    get {
+        guard let tokenResponse = self.authState.lastTokenResponse,
+              let token = tokenResponse.accessToken,
+              let tokenExp = tokenResponse.accessTokenExpirationDate,
+              tokenExp.timeIntervalSince1970 > Date().timeIntervalSince1970 else {
+                return nil
+        }
+        return token
+    }
+}
+```
+
+You can use [introspect](https://developer.okta.com/docs/api/resources/oidc/#introspect) endpoint to check the validity of the `accessToken`. This gives you more information about the token in the [response properties](https://developer.okta.com/docs/api/resources/oidc/#response-properties-3):
+
+```swift
+guard let accessToken = authStateManager?.accessToken else { return }
+
+authStateManager?.introspect(token: accessToken, callback: { payload, error in
+    guard let isValid = payload?["active"] as? Bool else {
+        // handle token non-valid case
+        return
+    }
+
+    // handle token valid case
+})
+```
+
+Since access tokens are traditionally short-lived, you can renew expired tokens by exchanging a refresh token for new ones:
```

Comment:
Since access tokens are traditionally short-lived, you can renew expired access tokens by exchanging a refresh token for new one: